packages/ra-ui-materialui/src/list/datagrid/DatagridRow.spec.tsx

```diff
@@ -51,9 +54,11 @@ describe('<DatagridRow />', () => {
     describe('rowClick', () => {
         it("should redirect to edit page if the 'edit' option is selected", () => {
             const { getByText, history } = renderWithRouter(
-                <DatagridRow {...defaultProps} rowClick="edit">
-                    <TitleField />
-                </DatagridRow>
+                <DatagridContextProvider value={dataGridContextValue}>
```

Comment:
You shouldn't need to add a context in your tests. It's a sign that you've broken backwards compatibility.

Instead, set the default context value to something that allows the component to work even when the developer hasn't defined the DatagridContext

Reply:
Done!

Reply:
You haven't actually addressed this - you still modify existing tests to make them pass, and it's the sign of a breaking change.